R/matrix_form.R

```diff
@@ -15,24 +15,36 @@ mform_handle_newlines <- function(matform) {
   has_topleft <- mf_has_topleft(matform)
   strmat <- mf_strings(matform)
   frmmat <- mf_formats(matform)
+  spamat <- mf_spans(matform)
+  alimat <- mf_aligns(matform)
   nr_header <- mf_nrheader(matform)
+  nl_inds_header <- seq(1, mf_nlheader(matform))
   hdr_inds <- 1:nr_header
 
   # hack that is necessary only if bottom aligned
   topleft_has_nl_char <- FALSE
   if (has_topleft) {
-    tl <- strmat[hdr_inds, 1, drop = TRUE]
-    strmat[hdr_inds, 1] <- ""
+    tl <- strmat[nl_inds_header, 1, drop = TRUE]
+    strmat[nl_inds_header, 1] <- ""
```

Comment:
This fix is needed when the number of rows are not the number of lines of the header (we need to get topleft out and add it back later to respect the right line grouping)